torch/csrc/jit/script/compiler.cpp

```diff
@@ -2108,6 +2109,68 @@ struct to_ir {
     } else {
       auto inputs = getNamedValues(apply.inputs(), true);
       auto attributes = emitAttributes(apply.attributes());
+
+      if (auto mv = std::dynamic_pointer_cast<MethodValue>(sv)) {
+        auto& callee = mv->getMethod();
+        try {
+          callee.ensure_defined();
+        } catch (std::exception&) {
+          throw ErrorReport(apply.range())
+              << " method '" << callee.name()
+              << "' is called recursively involving this call site. Recursive calls are not supported";
+        }
+
+        auto callee_graph = callee.graph();
+        auto mf = c10::ivalue::Function::create(callee_graph);
+        auto& schema = callee.getSchema();
+
+        if (mv->self()) {
+          inputs.insert(inputs.begin(), mv->self()->value(*callee_graph.get()));
+        }
+
+        std::stringstream failure_messages;
+        auto matched_schema = tryMatchSchema(
+            schema,
+            apply.range(),
+            *graph.get(),
+            c10::nullopt,
+            inputs,
+            attributes,
+            failure_messages,
+            /*conv_tensors_to_nums=*/true);
+
+        if (!matched_schema)
+        {
+          throw ErrorReport(apply.range())
+              << " mismatch between arguments and parameters " << failure_messages.str();
+        }
+
+        if (matched_schema.value().return_types.size() > 1)
+        {
+          throw ErrorReport(loc)
+              << "multiple returns aren't yet supported";
+        }
+
+        auto fun_constant = graph->insertConstant(mf, FunctionType::create(), apply.range());
+        auto call_node = graph->insertNode(create(prim::CallFunction, apply.range(), schema.returns().size()));
+
+        //add fun_constant
+        call_node->addInput(fun_constant);
+        auto& matched_inputs = matched_schema.value().inputs;
+        for (auto input : matched_inputs)
+        {
+          call_node->addInput(input);
+        }
+
+        call_node->eraseOutput(0);
```

Comment:
Following previous convention, I think it would be better if we added an `insertCallFunction(Value * callee, std::vector<Value*> inputs)` call to Graph. Like others, it should never do user-facing error reporting, but it should handle the details of what a CallFunction node looks like.